File: app/Models/Workflow.php
<?php

namespace App\Models;

use App\Models\TenantAwareModel;
use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Workflow extends TenantAwareModel
{
    use HasUuids, SoftDeletes;

    protected $fillable = [
        'tenant_id',
        'created_by',
        'name',
        'description',
        'status',
        'trigger_type',
        'trigger_config',
        'version',
    ];

    protected $casts = [
        'trigger_config' => 'array',
        'last_run_at' => 'datetime',
    ];

    public function isDraft(): bool    { return $this->status === 'draft'; }
    public function isActive(): bool   { return $this->status === 'active'; }
    public function isPaused(): bool   { return $this->status === 'paused'; }

    public function creator(): BelongsTo
    {
        return $this->belongsTo(User::class, 'created_by');
    }

    public function steps(): HasMany
    {
        return $this->hasMany(WorkflowStep::class)->orderBy('order');
    }

    public function executions(): HasMany
    {
        return $this->hasMany(Execution::class)->latest();
    }
}

File: mobile/lib/models/workflow.dart
class Workflow {
  final String id;
  final String name;
  final String? description;
  final String status;
  final String triggerType;
  final String? lastRunAt;
  final int version;

  const Workflow({
    required this.id,
    required this.name,
    this.description,
    required this.status,
    required this.triggerType,
    this.lastRunAt,
    required this.version,
  });

  factory Workflow.fromJson(Map<String, dynamic> json) => Workflow(
        id: json['id'] as String,
        name: json['name'] as String,
        description: json['description'] as String?,
        status: json['status'] as String,
        triggerType: json['trigger_type'] as String,
        lastRunAt: json['last_run_at'] as String?,
        version: (json['version'] as num).toInt(),
      );

  bool get isActive => status == 'active';
}
